sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -856,7 +857,31 @@ public <TContext> Flux<CosmosBulkOperationResponse<TContext>> executeBulkOperati
         return Flux.deferContextual(context -> {
             final BulkExecutor<TContext> executor = new BulkExecutor<>(this, operations, cosmosBulkExecutionOptions);
 
-            return executor.execute().publishOn(CosmosSchedulers.BULK_EXECUTOR_BOUNDED_ELASTIC);
+            return executor
+                .execute()
+                .doFinally((SignalType signal) -> {
```

Comment:
just a quick confirmation -- doFinally will implemented after the doOnComplete in bulkExecutor?